[FRT-879] Align Flink Artifact Terraform with CLI (#526)
confluentinc/terraform-provider-confluent: docs/resources/confluent_flink_artifact.md

The following arguments are supported:

> - `display_name` - (Required String) The Unique name of the Flink Artifact per cloud, region, environment scope.

Reviewer: Or is it intentional?

- `display_name` - (Required String) The unique name of the Flink Artifact per cloud, region, environment scope.
- `cloud` - (Required String) The cloud service provider that runs the Flink Artifact.
- `region` - (Required String) The cloud service provider region that hosts the Flink Artifact.

> - `class` - (Required String) Java class or alias for the Flink Artifact as provided by developer.

Reviewer 2: Let's not remove it outright, but rather mark it as deprecated.

```suggestion
- `class` - (Required String, , **Deprecated**) Java class or alias for the Flink Artifact as provided by developer.
```

Reviewer 2: also I don't have a full context here, but do we need to mark it as optional now?

In addition to the preceding arguments, the following attributes are exported:

- `id` - (Required String) The ID of the Flink Artifact, for example, `lfa-abc123`.

> - `content_format` - (Optional String) Archive format of the Flink Artifact (JAR or ZIP).
> - `runtime_language` - (Optional String) Runtime language of the Flink Artifact as Python or Java. The default runtime language is Java.

Reviewer 2: ```suggestion
- `content_format` - (Optional String) Archive format of the Flink Artifact. Accepted values are: `JAR`, `ZIP`.
- `runtime_language` - (Optional String) Runtime language of the Flink Artifact as `Python` or `Java`. Defaults to `Java`.
```

- `description` - (Optional String) Description of the Flink Artifact.

> - `documentation_link` - (Optional String) Documentation link of the Flink Artifact.

Author: Yes it was missing indeed
